src/libImaging/Convert.c

```diff
@@ -1209,6 +1217,8 @@ frompalette(Imaging imOut, Imaging imIn, const char *mode) {
         convert = alpha ? pa2l : p2l;
     } else if (strcmp(mode, "LA") == 0) {
         convert = alpha ? pa2la : p2la;
+    } else if (strcmp(mode, "P") == 0) {
+        convert = pa2p;
```

Comment:
Strange that Codecov doesn't think this is tested - when I run the test suite on my local machine, it is tested by [test_image_convert.py's test_sanity.](https://github.com/python-pillow/Pillow/blob/87ecd01fc042bb67b11bfcff3db99fc2b61e702d/Tests/test_image_convert.py#L14-L33)